### ISSUE TITLE
MD053 false positive

### ISSUE THREAD

**Issue by owenlamont** [rvben/rumdl#67]
Running rumdl fmt (v0.0.128) on the following markdown:

```markdown
# Enums

An attribute with a [class-private name] (beginning with, but not ending in, a double underscore) is
treated as a non-member:

[class-private name]: https://docs.python.org/3/reference/lexical_analysis.html#reserved-classes-of-identifiers

```

<img width="1755" height="289" alt="Image" src="https://github.com/user-attachments/assets/eb1b3502-41cd-4fb7-bd8d-69e78ecd5b34" />

Triggered rule [MD053](https://github.com/rvben/rumdl/blob/main/docs/md053.md) and removed the link... which clearly is referenced.

<img width="1066" height="119" alt="Image" src="https://github.com/user-attachments/assets/459262d5-7fb4-4c15-85bf-7257746ad728" />

leaving just:

```markdown
# Enums

An attribute with a [class-private name] (beginning with, but not ending in, a double underscore) is
treated as a non-member:
```

**Comment by rvben** [comment]
The removing of the whole item was way too aggressive and I removed fixing MD053 alltogether right now, as it seems hard to properly fix that automatically.
